clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/caching/view/model/KubernetesV2ServerGroup.java

```diff
@@ -193,13 +194,11 @@ public static KubernetesV2ServerGroup fromCacheData(KubernetesV2ServerGroupCache
     KubernetesManifestTraffic traffic = KubernetesManifestAnnotater.getTraffic(manifest);
     Set<String> explicitLoadBalancers =
         traffic.getLoadBalancers().stream()
-            .map(KubernetesManifest::fromFullResourceName)
-            .map(
-                p ->
-                    KubernetesManifest.getFullResourceName(
-                        p.getLeft(),
-                        p.getRight())) // this ensures the names are serialized correctly when
-            // the get merged below
+            // TODO(ezimanyi): Leaving this logic and the comment below, but I'm not sure that we
+            // still need the logic to parse then re-serialize the name.
+            // this ensures the names are serialized correctly when the get merged below
+            .map(lb -> KubernetesCoordinates.builder().fullResourceName(lb).build())
+            .map(c -> KubernetesManifest.getFullResourceName(c.getKind(), c.getName()))
```

Comment:
I am very confused...why do we need the second map?

Reply:
I am also very confused...because there is a comment `this ensures the names are serialized correctly when the get merged below` I was afraid to remove this.  This should be just an operation and its inverse...but presumably there was some condition where it wasn't an exact inverse which is why we need to do this?

That's why I left the comment because I was afraid of breaking some subtle use case and didn't want to spend more time digging into why we need these.  (The original PR adding it #3060 doesn't have any clues beyond that comment.)

Reply:
🆗